### DISCUSSION TITLE
error during build

### DISCUSSION THREAD

**Discussion by Skamiz** [Try/OpenGothic#263] [Q&A] [answered]
Hello, I am trying to get this running on Ubuntu according to the guide in the readme. But the last step 'make -C build' is interrupted by this error:
`
[  1%] Generating sprv/ssao_compose_rq.frag.sprv
/home/daren/github/OpenGothic/shader/ssao_compose.frag
WARNING: /home/daren/github/OpenGothic/shader/ssao_compose.frag:5: '#extension' : extension not supported: GL_EXT_ray_query
ERROR: /home/daren/github/OpenGothic/shader/ssao_compose.frag:20: '' :  syntax error, unexpected IDENTIFIER, expecting LEFT_BRACE or COMMA or SEMICOLON
ERROR: 1 compilation errors.  No code generated.

ERROR: Linking fragment stage: Missing entry point: Each stage requires one entry point
`
It's my understanding that I need to update some graphics library to make 'GL_EXT_ray_query' available. But I have no clue how to go about that.
Or maybe I am completely off.

Anyway, any help would be appreciated.

**Comment by Try** [comment]
Hi, @Skamiz !

Seems like your shader compiler is outdated. What is your version of `glslangValidator`?

**Comment by Skamiz** [comment, marked as answer]
daren@nuc01:~/github/OpenGothic$ glslangValidator -v
Glslang Version: 8.13.3559
ESSL Version: OpenGL ES GLSL 3.20 glslang Khronos. 13.3559
GLSL Version: 4.60 glslang Khronos. 13.3559
SPIR-V Version 0x00010400, Revision 1
GLSL.std.450 Version 100, Revision 1
Khronos Tool ID 8
SPIR-V Generator Version 8
GL_KHR_vulkan_glsl version 100
ARB_GL_gl_spirv version 100


I hope one of these lines answers your question, because I have little Idea what I am doing.

**Reply by Try** [reply]
> Glslang Version: 8.13.3559

Should be `10:11.7.0`, version 8 is too old

**Reply by Try** [reply]
Compiler is part of vulkan sdk, you can download latest vulkan sdk here: https://vulkan.lunarg.com/sdk/home#linux

**Reply by Skamiz** [reply]
seems to working now, many thanks

**Reply by Skamiz** [reply]
finished compiling and figured out the command to start the game. it runs at 8 - 20 fps, but that's on me for not having a graphics card, nonetheless is there a way to disable the dynamic shadows?

And also thank you a lot for letting me relive this piece of nostalgia. I don't think I had a finished G2 run in which I didn't cheat, so now I have the opportunity to correct that.

**Reply by Try** [reply]
You can set `shadowResolution` to zero, by modifying `render.h` - that for shadowmaps
Settings -> Game -> Cloud Shadows , to toggle SSAO effect

**Reply by Skamiz** [reply]
works, thank you :)
